### PR TITLE
Stacks archetype

### PR DESCRIPTION
**Pull request by pieper**
Should be good to go.  Could probably stand more testing but it works on data I have here.

### CODE REVIEW

**Comment by muratmaga** [comment]
I am getting this error:

```
Traceback (most recent call last):
  File "C:/Users/murat/Desktop/Slicer 4.13.0-2021-02-08/NA-MIC/Extensions-29693/SlicerMorph/lib/Slicer-4.13/qt-scripted-modules/ImageStacks.py", line 73, in setup
    self.fileModel.setTextElideMode(qt.ElideMiddle)
AttributeError: QStandardItemModel has no attribute named 'setTextElideMode'
```

**Comment by pieper** [comment]
Oops, thanks for testing.  Should be fixed now.  Was trying to make the file list show the end of the file path but Qt changed their way of doing it so there's no way to do it from python currently.  If you want to see the full path in the file list you can doubleclick on one of the items in the list.

**Comment by muratmaga** [comment]
Now I am getting this:
```
Switch to module:  "ImageStacks"
Traceback (most recent call last):
  File "C:/Users/murat/Desktop/Slicer 4.13.0-2021-02-08/NA-MIC/Extensions-29693/SlicerMorph/lib/Slicer-4.13/qt-scripted-modules/ImageStacks.py", line 278, in populateFromArchetype
    properties = self.logic.calculateProperties(filePaths)
  File "C:/Users/murat/Desktop/Slicer 4.13.0-2021-02-08/NA-MIC/Extensions-29693/SlicerMorph/lib/Slicer-4.13/qt-scripted-modules/ImageStacks.py", line 365, in calculateProperties
    reader.SetFileName(filePaths[0])
IndexError: list index out of range
Traceback (most recent call last):
  File "C:/Users/murat/Desktop/Slicer 4.13.0-2021-02-08/NA-MIC/Extensions-29693/SlicerMorph/lib/Slicer-4.13/qt-scripted-modules/ImageStacks.py", line 329, in onLoadButton
    outputNode = self.logic.loadByPaths(paths, self.currentNode(), properties)
  File "C:/Users/murat/Desktop/Slicer 4.13.0-2021-02-08/NA-MIC/Extensions-29693/SlicerMorph/lib/Slicer-4.13/qt-scripted-modules/ImageStacks.py", line 442, in loadByPaths
    path = paths[0]
IndexError: list index out of range
```
I just clicked a random slice within the folder with the select archaetype option.

**Comment by pieper** [comment]
What was the file pattern of your archetype?  In the help it tells which formats are supported.  The error you got is because there are no files in your file list.

I added some more checks to avoid the python errors.  I'm trying to think if there's a more obvious way to indicate that you need to populate the paths before you can load.  I guess disabling the load button until there are paths is the most obvious way.

**Comment by pieper** [comment]
Added the feature to disable the button when there are no paths present.

**Comment by muratmaga** [comment]
This was with the sample data we distribute with SlicerMorph:
https://github.com/SlicerMorph/SampleData/blob/master/sample_Skyscan_mCT_reconstruction.zip

They go left_side_damaged__rec0166.png, left_side_damaged__rec0167.png so forth.
I clicked a random one in the sequence.

**Comment by pieper** [comment]
Okay, that explains it - I'd been assuming they would always start at zero or one, since that's what I had here for test data, but starting with a random number is a special case I guess.  I wanted to avoid having entry boxes for start and end number.

So can we assume (insist) that the user select the lowest number in the sequence and that the files are sequentially numbered from there?

**Comment by pieper** [comment]
Okay, now this will let you pick something like image-00045.png and it will load everything from 45 to the last file that matches the extracted pattern in the directory.

My guess is that this covers most use case, but people sometimes get "creative" in file naming so we'll need to see if anything else pops up.

**Comment by muratmaga** [comment]
What if 0045 is not the first file? Do we expect/require them to click the very first slice in the list? I don't think people actually modify the filenames, it is the output of the reconstruction software...

**Comment by pieper** [comment]
Seems to be working for me but there could be special cases.  Let me know what you find.

**Comment by pieper** [comment]
Let's say the sequence is image-0045.png, image-0046.png, ... image-0104.png.  Currently they just need to select the first one in the series (the one with the lowest number, e.g. 0045 if that's the start of the sequence).  That seems like a natural thing to do.  I agree we don't want people to rename the files.

We could let them pick a file in the middle, say 0067, and then search backward to identify 0045 as the first in the sequence but that seems odd.  Currently is you pick 0067 you get 0067 through the end of the sequence.

**Comment by muratmaga** [comment]
Yes, it is working for me. But only if I choose the very first file in the stack as the archeatype, otherwise it starts from the selected file. I find this a bit confusing and somewhat limiting. If the goal is to constrain, they should use the selection dialog box anyways. I expected archetype to select everything that matches the format provided by sample file I provide, instead of assuming that I want it to start from there (that's what image sequences in Fiji does, and a lot of people would be familiar with that sort of usage. So is in all Bruker Skyscan software and similar ones that operate on image stacks).

In other words, you are not selecting where it starts from, you are simply selecting to present the filename convention to the parses (and assumption is to load everything in that folder that matches the pattern).

**Comment by pieper** [comment]
Okay, that's helpful to know.  I would have expected people to pick the first but I can change it.

**Comment by muratmaga** [comment]
The main issue is we do not know if they first file they are presented in the file dialog box is the first file in the stack (perhaps they are ordering by file size or date, or not even aware of it).

**Comment by pieper** [comment]
Okay, that was easy.  See if it seems more natural now.

There are other special cases we could think of that aren't currently handled.  Currently only the last number in the archetype (furthest right) is considered.  One can also type in your own sprintf pattern, but can't specify start and end numbering.  See if you think anything like that is worth supporting.

**Comment by muratmaga** [comment]
I think a regex based parsing can be useful, but not a priority. I see it as a more advanced issue. We might prioritize downsampling in the SkyscanReconImport as mentioned in #128

**Comment by pieper** [comment]
For now let's keep a list of the patterns we see in 'the wild' and then see what would be the easiest way for general users to specify them.
